BDArmory/Weapons/Missiles/MissileBase.cs

```diff
@@ -1629,11 +1636,54 @@ public void CheckDetonationState(bool separateWarheads = false)
                                             }
                                         }
                                     }
                                 }
+                                else if (TargetAcquired && targetVessel != null && targetVessel.Vessel != null)
+                                {
+                                    // For very high speed intercepts when missiles may phase through small vessels/missiles within a frame
+                                    Vector3 relPos = TargetPosition - vessel.CoM;
+                                    Vector3 relVel = TargetVelocity - vessel.Velocity();
+                                    bool approaching = Vector3.Dot(relPos, relVel) < 0f;
+                                    float targetRad = targetVessel.Vessel.GetRadius();
+                                    float selfRad = vessel.GetRadius();
+                                    float sepRad = (targetRad + selfRad);
+
+                                    if (approaching && relVel.sqrMagnitude * Time.fixedDeltaTime * Time.fixedDeltaTime > sepRad * sepRad)
+                                    {
+                                        try
+                                        {
+                                            Vector3 relAccel = TargetAcceleration - vessel.acceleration_immediate;
+                                            float cpaTime = AIUtils.TimeToCPA(relPos, relVel, relAccel, Time.fixedDeltaTime);
+
+                                            if (cpaTime < Time.fixedDeltaTime && cpaTime > 0f)
+                                            {
+                                                float sqrSep = AIUtils.PredictPosition(relPos, relVel, relAccel, cpaTime).sqrMagnitude;
+                                                if (sepRad * sepRad > sqrSep)
+                                                {
+                                                    // Valid hit, determine hit point and set transform.position (moves immediately) to hit point
+                                                    Vector3 missileToTarget = AIUtils.PredictPosition(TargetPosition, TargetVelocity, TargetAcceleration, cpaTime) - vessel.PredictPosition(cpaTime);
+                                                    if (Physics.Raycast(new Ray(TargetPosition - sepRad * missileToTarget, missileToTarget), out RaycastHit hit, sepRad * sepRad, (int)(LayerMasks.Parts | LayerMasks.EVA | LayerMasks.Wheels)))
+                                                        transform.position = hit.point;
+                                                    else // Just set position at CPA point
+                                                        transform.position = TargetPosition - missileToTarget;
+                                                    vessel.SetPosition(transform.position);
```

Comment:
This is setting the missile's position relative to where the target is now instead of where it would be in `cpaTime` seconds, which isn't actually where the CPA or hit is.
If nothing else is around, then this probably doesn't matter (which is likely the case 99.99% of the time), but if anything else were to be in the area where that would have happened, then those won't be being affected properly.
I suspect that that's about as good as it can get though without advancing the flight integrator by `cpaTime`, so maybe just add a comment to this for future reference saying that it's relative to the target's current position.